domain/src/Aggregates/SharePooling/SharePooling.php

```diff
@@ -99,14 +121,22 @@ public function applySharePoolingTokenDisposalReverted(SharePoolingTokenDisposal
     /** @throws SharePoolingException */
     public function disposeOf(DisposeOfSharePoolingToken $action): void
     {
-        if ($this->fiatCurrency && $this->fiatCurrency !== $action->proceeds->currency) {
+        if ($this->currencyMismatch($action->proceeds->currency)) {
             throw SharePoolingException::cannotDisposeOfFromDifferentCurrency(
                 sharePoolingId: $this->aggregateRootId,
                 from: $this->fiatCurrency,
                 to: $action->proceeds->currency,
             );
         }
 
+        if (! $action->isReplay() && $this->previousTransactionDate && $this->isOlderThanPreviousTransaction($action)) {
```

Comment:
ℹ️ Don't throw an exception when the disposal is being replayed, as its date will obviously be older than the current acquisition